onnxruntime/core/session/inference_session.cc

```diff
@@ -1372,6 +1355,17 @@
                                                                minimal_build_optimization_handling,
                                                                record_runtime_optimization_produced_op_schema));
 
+#ifdef USE_DML
+      if (execution_providers_.Get(kDmlExecutionProvider) && execution_providers_.NumProviders() <= 2) {
```

Comment:
>&& execution_providers_.NumProviders() <= 2 [](http://example.com/codeflow?start=58&length=43)

This shouldn't be necessary given the check for each node's provider type within the implementation of this transform